### DISCUSSION TITLE
arabic language problem

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5549] [Help]
hi everyone please help me
when i make the project in arabic the style of the project have somthing wrong beside after i choose arabic language when i click on another link it return to english and if i press on another link back to arabic how to avoid these problems?

follow are the steps:

![Screenshot (82)](https://user-images.githubusercontent.com/77009503/109495064-6e0e1880-7a97-11eb-9a2f-42e5920f94b0.png)

![Screenshot (83)](https://user-images.githubusercontent.com/77009503/109495075-723a3600-7a97-11eb-982d-347f84647fed.png)

![Screenshot (84)](https://user-images.githubusercontent.com/77009503/109495099-77978080-7a97-11eb-8f54-4a6bfc2492a0.png)

![Screenshot (86)](https://user-images.githubusercontent.com/77009503/109495194-98f86c80-7a97-11eb-913f-fe19f058fd2b.png)

**Comment by luongnv6** [comment]
Have you added the text file yet? 
![image](https://user-images.githubusercontent.com/21300748/109495400-ce19b480-7ac1-11eb-912e-2c159aa2e36e.png)

**Reply by samaremad** [reply]
no but i tried farsi language (which have file here in texts folder) it is also have the same issue when i click in a different list it return to previous language

**Reply by luongnv6** [reply]
try check in cookie
![image](https://user-images.githubusercontent.com/21300748/109502342-6f593880-7acb-11eb-8d8a-b9ebf0d14826.png)

**Reply by samaremad** [reply]
there are 3 cookies of Language Preference but all of them work in content : en

![Screenshot (87)](https://user-images.githubusercontent.com/77009503/109503613-2b523d80-7aa3-11eb-95a4-00e2d38bf107.png)

![Screenshot (88)](https://user-images.githubusercontent.com/77009503/109503636-3016f180-7aa3-11eb-8d25-930adbd00134.png)

![Screenshot (89)](https://user-images.githubusercontent.com/77009503/109503656-34430f00-7aa3-11eb-8c5e-37f169e1d868.png)

![Screenshot (90)](https://user-images.githubusercontent.com/77009503/109503674-386f2c80-7aa3-11eb-8fd5-c2b926099f5b.png)

**Reply by samaremad** [reply]
i delete these 3 cookies and try again first it work correctly but when i changed the language from page http://localhost:52719/Administration/Language

the proplem show again and i look in cookies and found 
that the 
http://localhost:52719/Administration/ 
added as en content but
http://localhost:52719/ 
add as ar content 
how can i avoid it from added 
http://localhost:52719/Administration 
to cookies

**Comment by samaremad** [comment]
hi everyone please help me with this issue

i have 2 language (English & Arabic)
so when the project load first in 1 language
then a cookie saved as localhost:35262/
then if i changed to another language and i am not on root as example if i am on localhost/adminstration/language
then i changed the language to arabic
then a new cookie will saved as localhost/adminstration/
with arabic content
so when i press on links in that path it will always convert me to arabic language even if in English mood

how can i solved this problem
plese help me it is urgent

**Reply by samaremad** [reply]
am i suppose to changed anything here ? or what can i do 

![155483206_447619969914154_494866734054829259_n](https://user-images.githubusercontent.com/77009503/109615157-3c9b5880-7b3c-11eb-8ffb-0b093ebb6a2a.jpg)

**Comment by samaremad** [comment]
i am thinking of writea function that check if that cookies allready existed then dont create another one 
but i dont know where to write that function
can any one help me with that?

**Comment by eissa1005** [comment]
Add this in class UserCultureProvider
 private static Dictionary<string, string> TwoLetterToFourLetter = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
        {
            { "ar", "ar-EG" },
            { "en", "en-US" },
            { "zh", "zh-CN" },
            { "vi", "vi-VN" },
            { "fa", "fa-IR" }
        };
===================
  private static List<CultureInfo> supportedCultures;
        private static readonly string[] supportedCultureIdentifiers = new string[] {
            "de-DE",
            "en-US",
            "en-GB",
            "es-ES",
            "fa-IR",
            "it-IT",
            "pt-PT",
            "pt-BR",
            "ru-RU",
            "tr-TR",
            "vi-VN",
            "zh-CN",
            "ar-EG"
        };
![lang1](https://user-images.githubusercontent.com/20481378/129072548-4251c807-930f-4d5f-bf98-63fa352ce378.png)
![lang2](https://user-images.githubusercontent.com/20481378/129072559-5f24c8cf-b6e1-454b-bfdb-1895f0478918.png)

**Reply by ra-bedo11** [reply]
فى خطا فى التاريخ
